src/server/scripts/Custom/SkipDK.cpp

```diff
@@ -41,6 +41,43 @@ constexpr auto OPTIONSKIPDK = 0;
 constexpr auto YESSKIPDK = 1;
 constexpr auto NOSKIPDK = 2;
 
+void Trinitycore_skip_deathknight_HandleSkip(Player* player)
+{
+    //Not sure where DKs were supposed to pick this up from, leaving as the one manual add
+    player->AddItem(6948, true); //Hearthstone
+
+    // these are all the starter quests that award talent points, quest items, or spells
+    int STARTER_QUESTS[39] = { 12593, 12619, 12842, 12848, 12636, 12641, 12657, 12678, 12679, 12680, 12687, 12698, 12701, 12706, 12716, 12719, 12720, 12722, 12724, 12725, 12727, 12733, 12739, 12742, 12743, 12744, 12745, 12746, 12747, 12748, 12749, 12750, 12751, 12754, 12755, 12756, 12757, 12779, 12801 };
+
+    for (auto questId : STARTER_QUESTS)
+    {
+        if (player->GetQuestStatus(questId) == QUEST_STATUS_NONE)
+        {
+            player->AddQuest(sObjectMgr->GetQuestTemplate(questId), nullptr);
+            player->RewardQuest(sObjectMgr->GetQuestTemplate(questId), false, player);
+        }
+    }
+
+    int DKL = sConfigMgr->GetFloatDefault("Skip.Deathknight.Start.Level", 58);
+    if (player->GetLevel() <= DKL)
+    {
+        //GiveLevel updates character properties more thoroughly than SetLevel
+        player->GiveLevel(DKL);
+    }
```

Comment:
This was something being debated as well of either Setlevel and GiveLevel by someone else before and i was told a Pr was going to be put incoporating it... but that was over 6 months ago and never saw the pr yet. So i didn't wanted to make that change as i did not wanted to have the impression of yanking someone's suggestion with out their appropiate gitname to credit for the suggestion.

Reply:
Apologies to whoever had this idea originally, I think it is correct though since GiveLevel properly checks things like glyphs and stats.

Reply:
I have no issues of u doing it. As it was discussed privately to me from the other person. So you making mention of it in  a git discuss ensures it was infact a original public idea on your end.